frontend/src/v5/store/teamspaces/teamspaces.helpers.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { clientConfigService } from '@/v4/services/clientConfig';
+import { generateV5ApiUrl } from '@/v5/services/api/default';
+import { fetchTeamspaceAvatar } from '@/v5/services/api/teamspaces';
+
+export const getTeamspaceAvatarUrl = async (teamspace) => {
+	const DEFAULT_AVATAR_URL = 'assets/images/teamspace_placeholder.svg';
+	try {
+		await fetchTeamspaceAvatar(teamspace);
+		return generateV5ApiUrl(`teamspaces/${teamspace}/avatar?${Date.now()}`, clientConfigService.GET_API);
+	} catch (e) {
+		return DEFAULT_AVATAR_URL;
+	}
+};
```

Comment:
Instead of doing this I think is easier to deal with it in the component

```
const [src, setSrc] = useState(props.src); 
const onErrorImg = (event) => {
		setSrc(props.defaultSrc);
};

return (<img src={src} onerror="onError" />);
```